app/scripts/app-init.js

```diff
@@ -111,14 +99,13 @@ function importAllScripts() {
   }
 }
 
-// eslint-disable-next-line no-undef
-self.addEventListener('install', importAllScripts);
+importAllScripts();
 
 /*
  * Message event listener below loads script if they are no longer available.
  * chrome below needs to be replaced by cross-browser object,
  * but there is issue in importing webextension-polyfill into service worker.
  * chrome does seems to work in at-least all chromium based browsers
  */
-// eslint-disable-next-line no-undef
-chrome.runtime.onMessage.addListener(importAllScripts);
+// eslint-disable-next-line
+chrome.runtime.onMessage.addListener(() => {});
```

Comment:
After the latest changes around `importAllScripts`, can we remove this line and the comments above?

Reply:
reactivating service worker does not work if I remove this line

Reply:
you need to add a comment explaining that the otherwise meaningless looking line is important

Reply:
```js
chrome.runtime.onMessage.addListener((message) => {
  return message.name === WORKER_KEEPALIVE_MESSAGE;
});
```
Something like this, while useless, would accomplish following things:
1. share a reference to the message name so that it's easy to navigate the codebase to find where the keepalive is sent
2. explain what the line is for
3. avoid future breakage if a browser decides to unregister/ignore empty listeners to prevent unnecessary keepalive. (If you find that impossible, read up on the beforeunload event)

Reply:
ah makes sense now. I didn't realize there was a dual purpose in the previous iteration. We need to add the listener on the service worker to receive the messages for  `WORKER_KEEP_ALIVE_MESSAGE` 👍  

Thanks for updating the comment. I'm also a fan of @naugtur's comment and code to receive the message in hand.

Reply:
ah makes sense now. I didn't realize there was a dual purpose in the previous iteration. We need to add the listener on the service worker to receive the messages for  `WORKER_KEEP_ALIVE_MESSAGE` 👍  

Thanks for updating the comment! I'm also a fan of @naugtur's comment and code to receive the message in hand.

Reply:
I made the update